pyth_observer/dispatch.py

```diff
@@ -62,8 +79,41 @@ async def run(self, states: List[State]):
 
                 sent_events.append(event.send())
 
+                if event_type == "ZendutyEvent":
+                    # Add failed check to open alerts
+                    alert_identifier = (
+                        f"{check.__class__.__name__}-{check.state().symbol}"
+                    )
+                    state = check.state()
+                    if isinstance(state, PublisherState):
+                        alert_identifier += f"-{state.publisher_name}"
+                    self.open_alerts[alert_identifier] = datetime.now().isoformat()
+
         await asyncio.gather(*sent_events)
 
+        # Check open alerts and resolve those that are older than 2 minutes
+        if "ZendutyEvent" in self.config["events"]:
+
+            # Write open alerts to file to ensure persistence
+            with open(self.open_alerts_file, "w") as file:
+                json.dump(self.open_alerts, file)
```

Comment:
Is there a specific reason that you write before modifying it? I think writing after removing the older ones might be better.